shell/packages/accounts-email-token/token_server.js

```diff
@@ -53,8 +57,9 @@ Accounts.registerLoginHandler("emailToken", function (options) {
     };
   }
 
-  if (!user.services || !user.services.emailToken ||
-      !(user.services.emailToken.tokens)) {
+  var identity = user.identities[0];
```

Comment:
The projection `{fields: {"identities.$": 1}` will only populate the `identities` array with the first element that matched the query. http://docs.mongodb.org/manual/reference/operator/projection/positional/#proj._S_